src/vs/workbench/contrib/webview/browser/pre/main.js

```diff
@@ -491,8 +491,13 @@
 				const onLoad = (contentDocument, contentWindow) => {
 					if (contentDocument && contentDocument.body) {
 						// Workaround for https://github.com/Microsoft/vscode/issues/12865
-						// check new scrollY and reset if necessary
-						setInitialScrollPosition(contentDocument.body, contentWindow);
+						// check new scrollY and reset if neccessary
+
+						// Delay this in the hope that all of the iframe content will have loaded by the time we set
+						// the desired initial scroll position. But how can we be sure how long is long enough?
+						setTimeout(() => {
```

Comment:
I don't think we should change this to fix this issue. This code is run for all webviews changing it may break other webviews

Can you tell me more about why was this part of the change is needed?

Reply:
I take your point about this change having effects beyond this issue, but I found that the onLoad in which I made this change is getting called even before the whole of the iframe content has loaded. I commented on this earlier in this PR, e.g. https://github.com/microsoft/vscode/pull/85982#issuecomment-562985969 and https://github.com/microsoft/vscode/pull/85982#issuecomment-573640635

Reply:
This is by-design. We don't want to block the entire webview from loading if it happens to load a slow resource loaded inside of it

What happens if you remove this part of the change?

Reply:
As I recall, without this part of the change the restore of scroll position is unreliable. You get scrolled somewhere, but not necessarily back to exactly the position you were at. If an extension's README.md includes some large screenshots for example. For resetting the former scroll position accurately it seems to me to be necessary for the webview to wait until the resources have loaded. Or at least all of the resources that occupy space above the desired scroll position.

Reply:
Again, doing this for all webviews seems like it will cause other problems. It's also a poor experience to have the scroll position suddenly jump when all the images are loaded, so maybe there's a better compromise here

Reply:
I'm running out of ideas. My understanding of the webviews code is that the mechanism to (optionally) restore the previous scroll position when reviving(?) a webview tries to set it to the same percentage (`initData.initialScrollProgress`) of the entire document contents as the former scroll position was at. Assuming that at the time the old scroll position was noted all the image resources had been fully loaded, then unless the same is true at the time we reinstate the scroll position for the revived document there's a chance we won't put the user back to where they remember being when they switched away.

So I'm going to remove my main.js change, leaving an imperfect fix for the original issue. But the imperfection will probably only show up in extension READMEs that pull in big images.